packages/salesforcedx-vscode-core/src/settings/pushOrDeployOnSave.ts

```diff
@@ -9,99 +9,55 @@ import { channelService } from '../channels';
 import { getWorkspaceOrgType, OrgType } from '../context';
 import { nls } from '../messages';
 import { notificationService } from '../notifications';
+import { IsInSfdxPackageDirectory } from '../predicates';
 import { sfdxCoreSettings } from '../settings';
-
 import { SfdxProjectJsonParser } from '../util';
 
 import * as fs from 'fs';
 import * as path from 'path';
 import * as vscode from 'vscode';
 
-export enum FileEventType {
-  Create,
-  Change,
-  Delete
-}
-
-const WAIT_TIME_IN_MS = 50;
+const WAIT_TIME_IN_MS = 4500;
```

Comment:
I played around with different wait times and settled on 4500ms, but I am very open to suggestions for other wait times. In my own development, I don't tend to make a lot of rapid changes within 4.5 seconds of each other, so my way of development will still trigger many deploys, but when I extended the wait time much longer than 4.5 seconds, the time between my last change and the deployment felt like a noticeable lag.

Reply:
Sounds good. We could consider making it configurable if it’s not too difficult, but 4500 seems like a good default.